src/index.ts

```diff
@@ -94,10 +123,19 @@ export async function encryptWithKey<R>(
   const buffer = new Uint8Array(buf);
   const vectorStr = Buffer.from(vector).toString('base64');
   const vaultStr = Buffer.from(buffer).toString('base64');
-  return {
+  const encryptionResult: EncryptionResult = {
     data: vaultStr,
     iv: vectorStr,
   };
+
+  if (
+    !(encryptionKey instanceof CryptoKey) &&
+    encryptionKey.derivationFunction
+  ) {
+    encryptionResult.keyMetadata = encryptionKey.derivationFunction;
+  }
+
+  return encryptionResult;
```

Comment:
Because if the user encrypts data using an arbitrary `CryptoKey` we can't assume that it was derived with this library, thus we should leave the metadata empty